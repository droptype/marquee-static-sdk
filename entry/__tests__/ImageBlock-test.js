const React = require('react')
const ImageBlock = require('../ImageBlock')
const renderer = require('react-test-renderer')

const SAMPLE_BLOCK = require('./SAMPLE_ENTRY.json').content.find( block => 'image' === block.type)

describe('ImageBlock', () => {
    it('renders', () => {
        const component = renderer.create(
            React.createElement(ImageBlock, { block: SAMPLE_BLOCK })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })
    it('renders plain', () => {
        const component = renderer.create(
            React.createElement(ImageBlock, { block: SAMPLE_BLOCK, plain: true })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })
})
