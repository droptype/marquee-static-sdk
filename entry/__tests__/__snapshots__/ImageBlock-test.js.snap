// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBlock renders 1`] = `
<figure
  className={
    Classes {
      "_groups": Object {
        "position": "right",
        "size": "medium",
      },
      "_initial": Array [
        "Block",
        "ImageBlock",
      ],
      "_switches": Object {},
    }
  }
  id="image_fe7b8a48f94b4ccb95a38beb91d00ff7"
>
  <div
    className="_Content"
  >
    <img
      alt="The spire of the Crysler building in the background, the tops of various skyscrapers in front of it."
      data-src_128="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/saO09kR7TO6Fdkmpdo60_DSC_6110.jpg"
      data-src_1280="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/GLYe0WxSlem5uDSNxJ4M_DSC_6110.jpg"
      data-src_2560="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/jKQgkPtRgSiIUoKbdoIc_DSC_6110.jpg"
      data-src_640="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg"
      sizes="(min-width: 720px) 720px,100vw"
      src="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg"
      srcSet="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/saO09kR7TO6Fdkmpdo60_DSC_6110.jpg 128w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg 640w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/GLYe0WxSlem5uDSNxJ4M_DSC_6110.jpg 1280w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/jKQgkPtRgSiIUoKbdoIc_DSC_6110.jpg 2560w"
    />
    <figcaption
      className="_Caption"
    >
      <p
        className="_CaptionText"
      >
        The Crysler building.
      </p>
      <p
        className="_Credit"
      >
        Alec Perkins
      </p>
    </figcaption>
  </div>
</figure>
`;

exports[`ImageBlock renders plain 1`] = `
<figure>
  <img
    src="https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg"
  />
  <figcaption
    className={null}
  >
    <p
      className={null}
    >
      The Crysler building.
    </p>
    <p
      className={null}
    >
      Alec Perkins
    </p>
  </figcaption>
</figure>
`;
